src/analysis/functions.rs

```diff
@@ -238,51 +260,201 @@ fn analyze_function(
+                        &mut commented,
+                        imports,
+                        &c_parameters,
+                    ) {
+                        if let Some(destroy_index) = destroy_index {
+                            let user_data = cross_user_data_check.entry(destroy_index)
+                                                                 .or_insert_with(|| callback.user_data_index);
+                            if *user_data != callback.user_data_index {
+                                error!("`{}`: Different destructors cannot share the same user data",
+                                       func.name);
+                                commented = true;
+                            }
+                            callback.destroy_index = destroy_index;
+                        } else {
+                            user_data_indexes.insert(callback.user_data_index);
+                            to_remove.push(callback.user_data_index);
+                        }
+                        callbacks.push(callback);
+                        to_replace.push((pos, par.typ));
+                        continue;
+                    }
+                } else if par.c_type == "GDestroyNotify" {
+                    if let Some((mut callback, _)) = analyze_callback(
+                        func_name,
+                        env,
+                        &par,
+                        &callback_info,
+                        &mut commented,
+                        imports,
+                        &c_parameters,
+                    ) {
+                        // We just assume that for API "cleaness", the destroy callback will always be
+                        // |-> *after* <-| the initial callback.
+                        if let Some(user_data_index) = cross_user_data_check.get(&pos) {
+                            callback.user_data_index = *user_data_index;
+                            callback.destroy_index = pos;
+                        } else {
+                            error!("`{}`: no user data point to the destroy callback", func_name);
+                            commented = true;
+                        }
+                        destroys.push(callback);
+                        to_remove.push(pos);
+                        continue;
+                    }
+                }
+                if !commented {
+                    commented |= parameter_rust_type(env, par.typ, par.direction, Nullable(false), RefMode::None).is_err();
+                }
+            }
+        }
+
+        // Check for cross "user data".
+        if cross_user_data_check.values().collect::<Vec<_>>().windows(2).any(|a| a[0] == a[1]) {
             commented = true;
+            error!("`{}`: Different user data share the same destructors", func.name);
         }
-    }
 
-    if async && trampoline.is_none() {
-        commented = true;
+        if expecting_data && (!destroys.is_empty() || !callbacks.is_empty()) {
```

Comment:
You probably want to print something to the user if `expecting_data` but we have no destroys or callbacks

Reply:
Also the whole function should be ignored then I guess?